lib/json_api_controller/precondition_check.rb

```diff
@@ -0,0 +1,28 @@
+module JsonApiController
+  module PreconditionCheck
+    def precondition_check
+      if !precondition
+        raise PreconditionNotPresent, precondition_error_msg
+      elsif precondition_fails?
+        raise PreconditionFailed
+      end
+    end
+    
+    def precondition
+      case action_name
+      when "show"
+        request.headers["If-Modified-Since"]
+      else
+        request.headers["If-Unmodified-Since"]
```

Comment:
No it's not the right header in the issue, I hadn't finished reading the precondition spec when I pitched it. I documented the correct header in Apiary.